step-templates/hashicorp-vault-keyvalue-v2-retrieve-secrets.json

```diff
@@ -0,0 +1,65 @@
ret = [PsCustomObject]@{\n            EnginePath   = $enginePath.Trim('/')\n            Path         = $path.Trim('/')\n            FieldName    = $field\n            VariableName = if (![string]::IsNullOrWhiteSpace($item[1])) { $item[1].Trim() } else { \"\" }\n        }\n        return $secret\n    }\n}\n\n$SecretsSpecified = ($Secrets.Count -gt 0)\nif ($SecretsSpecified -eq $False) {\n    throw \"No secrets could be derived from parameter values!\"\n}\n\nWrite-Verbose \"VAULT_RETRIEVE_KV_V2_SECRETS_ADDRESS: $VAULT_RETRIEVE_KV_V2_SECRETS_ADDRESS\"\nWrite-Verbose \"VAULT_RETRIEVE_KV_V2_SECRETS_API_VERSION: $VAULT_RETRIEVE_KV_V2_SECRETS_API_VERSION\"\nWrite-Verbose \"VAULT_RETRIEVE_KV_V2_SECRETS_TOKEN: '********'\"\nWrite-Verbose \"VAULT_RETRIEVE_KV_V2_SECRETS_PATH: $VAULT_RETRIEVE_KV_V2_SECRETS_PATH\"\n\ntry {\n    $VariablesCreated = 0\n   \n    foreach ($Secret in $Secrets) {\n        $WorkingPath = \"$($Secret.EnginePath)/$($Secret.Path)\"\n        $RequestPath = \"$($Secret.EnginePath)/data/$($Secret.Path)\"\n        $uri = \"$VAULT_RETRIEVE_KV_V2_SECRETS_ADDRESS/$VAULT_RETRIEVE_KV_V2_SECRETS_API_VERSION/$RequestPath\"\n        $headers = @{\"X-Vault-Token\" = $VAULT_RETRIEVE_KV_V2_SECRETS_TOKEN }\n\n        Write-Verbose \"Making GET request to $uri\"\n        $response = Invoke-RestMethod -Uri $uri -Headers $headers -Method GET\n\n        if ($null -ne $response) {\n            $fieldName = $Secret.FieldName\n            $value = $response.data.data.$fieldName\n            $variableName = $Secret.VariableName\n            if ($null -ne $value) {\n                Write-Verbose \"Found matching field for $($Secret.FieldName)\"\n                if ([string]::IsNullOrWhiteSpace($variableName)) {\n                    $variableName = \"$($WorkingPath.Replace(\"/\",\".\")).$($Secret.FieldName)\"\n                    Write-Verbose \"No output variable name specified for $($Secret.FieldName). Using generated name of '$($variableName)'.\"\n                }\n                \n                Write-Verbose \"Setting output variable: $($variableName)\"\n                Set-OctopusVariable -Name $variableName -Value $value -Sensitive\n                $VariablesCreated += 1\n            }\n        }  \n        else {\n            Write-Error \"Null or Empty response returned from path: $WorkingPath\" -Category InvalidResult\n        } \n    }\n}\ncatch {\n    $ExceptionMessage = $_.Exception.Message\n    $ErrorBody = Get-WebRequestErrorBody -RequestError $_\n    $Message = \"An error occurred retrieving secrets from path: $WorkingPath - $ExceptionMessage\"\n    if ($null -ne $ErrorBody.errors) {\n        $AdditionalDetail += $ErrorBody.errors -Join \",\"\n    }\n    else {\n        $AdditionalDetail += $ErrorDetails\n    }\n    if (![string]::IsNullOrWhiteSpace($AdditionalDetail)) {\n        $Message += \"`n`tDetail: $AdditionalDetail\"\n    }\n\n    Write-Error $Message -Category ReadError\n}\n\nWrite-Host \"Created $variablesCreated output variables\""
```

Comment:
Same thing here, use write-highlight to let people know what the name of the output variable is so they can use it later.

Reply:
Added in https://github.com/OctopusDeploy/Library/commit/517ee0900d37c8c4475fc1ec8e90592bebbceaab